invenio_indexer/api.py

```diff
@@ -51,7 +51,8 @@ class RecordIndexer(object):
     """
 
     def __init__(self, search_client=None, exchange=None, queue=None,
-                 routing_key=None, version_type=None, record_to_index=None):
+                 routing_key=None, version_type=None, record_to_index=None,
+                 record_cls=Record):
```

Comment:
I'm a bit conflicted regarding if this should be in the `__init__` or just as a class member to allow for easier sub-classing:

```python
class RecordIndexer:

    record_cls = Record
    ...

# to subclass:
class CommunityIndexer(RecordIndexer):

    record_cls = CommunityRecord
```

**vs.**

```python
CommunityIndexer = partial(RecordIndexer, record_cls=CommmunityRecord)
```

Reply:
To be honest I would prefer to subclass it since it seems seamless and more explicit.